telegram-translation-shortcuts.user.js

```diff
@@ -70,16 +70,53 @@ function scrollItems (down) {
  * Clicks the 'Add Translation' button
  */
 function addTranslation () {
-  document.getElementsByClassName('key-add-suggestion-header').item(0).click()
+  // don't click if input box is expanded
+  var input_wrapper_collapsed = (document.getElementsByClassName('key-add-suggestion-wrap')[0].className == 'key-add-suggestion-wrap');
+  
+  if (!input_wrapper_collapsed)   // don't toggle 'add' and 'cancel'
+    document.getElementsByClassName('key-add-suggestion-header').item(0).click() 
+  else
+    document.getElementsByClassName('input form-control tr-form-control key-add-suggestion-field').item(0).focus() //fixme: throws undefined error
 }
 
 /**
  * Clicks the edit icon
  */
 function editTranslation () {
-  document.getElementsByClassName('ibtn key-suggestion-edit').item(0).click()
+  //TODO: don't click if input box is expanded
+  document.getElementsByClassName('ibtn key-suggestion-edit').item(0).click()  
 }
 
+/**
+ * Clicks the cancel button
+ */
+function cancelTranslation () {
+  // restrict when cancel button can be pressed
+  if (document.getElementsByClassName('key-add-suggestion-wrap')[0].className == 'key-add-suggestion-wrap') {
+    document.getElementsByClassName('btn btn-default form-cancel-btn').item(0).focus();
+    document.getElementsByClassName('btn btn-default form-cancel-btn').item(0).click();
+  }
+}
+
+/**
+ * Clicks the submit button
+ */
+function submitTranslation () {
+  // restrict when submit button can be pressed
+  if (document.getElementsByClassName('key-add-suggestion-wrap')[0].className == 'key-add-suggestion-wrap') {
+    document.getElementsByClassName('btn btn-primary form-submit-btn').item(0).focus()
+    document.getElementsByClassName('btn btn-primary form-submit-btn').item(0).click()
+  }
+}
+
+/**
+ * Clicks the delete icon
+ */
+/*
+function deleteTranslation () {
```

Comment:
Why is this function commented out? Do you plan to use it in the future?

Reply:
Yeah, but this requires some code for selecting suggestions. I thought about CTRL+1,2,3... for selecting. From there, Apply, Edit and Delete can be used for each suggestion item. Not every user can apply/delete the suggestions, so that needs proper detection as well. The alternative would be CTRL+Up/Down to select suggestions.

Up/Down is already used for scrolling through search results.